src/System.DirectoryServices.Protocols/tests/LdapSessionOptionsTests.cs

```diff
@@ -609,10 +609,11 @@ public void StartTransportLayerSecurity_Disposed_ThrowsObjectDisposedException()
         }
 
         [Fact]
-        public void StopTransportLayerSecurity_NotStarted_ThrowsTlsOperationException()
+        public void StopTransportLayerSecurity_NotStarted_ThrowsException()
         {
             LdapSessionOptions options = new LdapConnection("server").SessionOptions;
-            Assert.Throws<TlsOperationException>(() => options.StopTransportLayerSecurity());
+            Exception exception = Assert.ThrowsAny<Exception>(() => options.StopTransportLayerSecurity());
+            Assert.True(exception is TlsOperationException || exception is ObjectDisposedException, $"Unexpected exception of type {exception.GetType()} was thrown.");
```

Comment:
Was the failure only in a release build?  It sounds like the LdapConnection got finalized, and that the right fix is just to keep the LdapConnection object referenced until after the StopTransportLayerSecurity call.